iOS_Swift/PERSISTANCE.md

```diff
@@ -0,0 +1,152 @@
+## Persistance in iOS
+
+### Why 
+
+Sometimes in an iOS App you need to store some data in the users hard drive so it can **persist** among app launches, hence the name, **persistance.**
+
+This can be something as simple as cache so that the user has a faster experience and doesn't have to make more extra requests to get data.
+
+This can also be the settings or the session if your app has a login component.
+
+Or it can be any kind of app state that you need to store. This can all be acomplished in several ways, each has it's benefits and different complexity levels. I'll try to go trough all from the simplest to the most complex.
+
+### NSUserDefaults
+
+The simplest way to store small amounts of data like user settings or session information is in the User Defaults. **NSUserDefaults** is a singleton class which already handles caching and persisting anything you add to it. You just need to use a string **key** and store any kind of object in it. It is basically a key-value store, that handles caching and on-disk-persistance on the fly for you.
```

Comment:
Not any kind of object:

> A default object must be a property list, that is, an instance of (or for collections a combination of instances of): NSData, NSString, NSNumber, NSDate, NSArray, or NSDictionary" - [NSUserDefault Class Reference](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSUserDefaults_Class/)

It is possible to encode your object into a NSData object but if your find yourself doing that it is probable that another method of persistence is more appropriate. Ex. saving object as a file in disk